packages/rpc-types/src/lamports.ts

```diff
@@ -46,12 +49,28 @@ export function lamports(putativeLamports: bigint): LamportsUnsafeBeyond2Pow53Mi
     return putativeLamports;
 }
 
+type ExtractAdditionalProps<T, U> = Omit<T, keyof U>;
+
 export function getLamportsEncoder(): FixedSizeEncoder<LamportsUnsafeBeyond2Pow53Minus1, 8> {
     return getMemoizedU64Encoder();
```

Comment:
super nit: maybe also wrap that one in the helper function that accepts an inner encoder for consistency?